From 0751453581235200615be0bd8ffd10fe2b5a9ed1 Mon Sep 17 00:00:00 2001
From: xuxiang <xuxiang@eswincomputing.com>
Date: Sun, 9 Jun 2024 16:37:28 +0800
Subject: [PATCH 082/109] WIN2030-15279:fix:i2c1 bus err after mcu key rst

Changelogs:
1. It failed to read eeprom after mcu key reset
2. add func eeprom read failed retry

Change-Id: Ifd954f9e4295efa1b8395e7439510ea868e2c160
---
 Core/Src/main.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/Core/Src/main.c b/Core/Src/main.c
index fb863ac..72619d1 100644
--- a/Core/Src/main.c
+++ b/Core/Src/main.c
@@ -131,13 +131,20 @@ int main(void)
   */
 extern void hf_power_task (void* parameter);
 extern void hf_gpio_task (void* parameter);
-
+int retry_count = 10;
 void hf_main_task(void *argument)
 {
+  int ret = 0;
   printf("HiFive 106SC!\n");
 
   /* get board info from eeprom where the MAC is stored */
-  if(es_init_info_in_eeprom()) {
+  do
+  {
+    osDelay(220);
+    ret = es_init_info_in_eeprom();
+  } while (ret && (retry_count--));
+  
+  if(ret || retry_count <=0) {
     printf("severe error: get info from eeprom failed!!!");
     while(1);
   }
-- 
2.25.1

